### ISSUE TITLE
Implement Convert<HashMap<Value, Value>>

### ISSUE THREAD

**Issue by lopopolo** [artichoke/artichoke#29]
Once artichoke/artichoke#28 lands, `Value` is a valid key for a `HashMap`.

Implementing `Convert<HashMap<Value, Value>>` allows us to follow the same approach as `Array` and massively simplify the macro madness in [`hash.rs`](https://github.com/lopopolo/ferrocarril/blob/329c544006b619b7b4fac9c96b6b5ec51a311a84/mruby/src/convert/hash.rs).

I think this should also unify the special-cased `&str` converters.

**Comment by lopopolo** [comment]
Closing as WONTFIX. `HashMap<Value, Value>` cannot be constructed because `Value` cannot implement `Hash` and `PartialEq`. See rationale in https://github.com/artichoke/artichoke/issues/28#issuecomment-757421703.
